File: front/frontclient/src/app/game.ts
import { FIGURE } from "./figure";
import { Movement } from "./movement";
import { PLAYER } from "./player";

export class Game {
    id: number | undefined;
    user: string | undefined;
    winner: boolean | undefined;
    winnerPlayer: PLAYER | null;
    movements: Movement[];
    tuple?: { player: FIGURE, machine: FIGURE }[];

    constructor() {
        this.id = undefined;
        this.user = undefined;
        this.winner = undefined;
        this.winnerPlayer = null;
        this.movements = [];
    }
}

File: src/main/kotlin/games/rockpaperscissors/model/Game.kt
package games.rockpaperscissors.model

import jakarta.persistence.*


@Entity
@Table(name = "game")
data class Game (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,
    @Column(name = "user_name", nullable = false)
    val user: String,
    @Column
    var winner: Boolean = false,
    @Column
    var winnerPlayer: Player? = null,
    @OneToMany(mappedBy = "game_id")
    val movements: List<Movement> = emptyList()
)